server/rtcd.go

```diff
@@ -177,27 +438,94 @@ func (p *Plugin) getRTCDClientConfig(rtcdURL string) (rtcd.ClientConfig, error)
 
 	// Here we need some coordination to avoid multiple plugin instances to
 	// register at the same time (at most one would succeed).
-	mutex, err := cluster.NewMutex(p.API, "rtcd_registration")
+	mutex, err := cluster.NewMutex(m.ctx.API, "rtcd_registration")
 	if err != nil {
 		return cfg, fmt.Errorf("failed to create cluster mutex: %w", err)
 	}
 
-	lockCtx, cancelCtx := context.WithTimeout(context.Background(), 5*time.Second)
+	lockCtx, cancelCtx := context.WithTimeout(context.Background(), lockTimeout)
 	defer cancelCtx()
 	if err := mutex.LockWithContext(lockCtx); err != nil {
 		return cfg, fmt.Errorf("failed to acquire cluster lock: %w", err)
 	}
 	defer mutex.Unlock()
 
-	if storedCfg, err := p.getStoredRTCDConfig(); err != nil {
+	if storedCfg, err := m.getStoredRTCDConfig(); err != nil {
 		return cfg, fmt.Errorf("failed to get rtcd credentials: %w", err)
 	} else if storedCfg.URL == cfg.URL && storedCfg.ClientID == cfg.ClientID {
 		return storedCfg, nil
 	}
 
-	if cfg, err := p.registerRTCDClient(cfg); err != nil {
+	if cfg.AuthKey == "" {
+		m.ctx.LogDebug("auth key missing from rtcd config, generating a new one")
+		cfg.AuthKey, err = random.NewSecureString(32)
+		if err != nil {
+			return cfg, fmt.Errorf("failed to generate auth key: %w", err)
+		}
+	}
```

Comment:
Forgot to mention that this PR also includes the auth related changes proposed in https://github.com/mattermost/rtcd/pull/28 . This simplifies a bit and removes the corner case of getting locked out due to crashing after registration but before saving to the store. We now save the key prior to registration in `registerRTCDClient`.